core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -376,22 +376,29 @@ public Catalog createCatalog(
 
       store.put(e, false /* overwrite */);
       CatalogWrapper wrapper = catalogCache.get(ident, id -> createCatalogWrapper(e));
-      createSuccess = true;
+
+      needClean = false;
       return wrapper.catalog;
+
     } catch (EntityAlreadyExistsException e1) {
+      needClean = false;
       LOG.warn("Catalog {} already exists", ident, e1);
       throw new CatalogAlreadyExistsException("Catalog %s already exists", ident);
+
     } catch (IllegalArgumentException | NoSuchMetalakeException e2) {
       throw e2;
+
     } catch (Exception e3) {
       catalogCache.invalidate(ident);
       LOG.error("Failed to create catalog {}", ident, e3);
       if (e3 instanceof RuntimeException) {
         throw (RuntimeException) e3;
       }
       throw new RuntimeException(e3);
+
     } finally {
-      if (!createSuccess) {
+      if (needClean) {
+        // Clean up the default schema under the catalog if creation failed
```

Comment:
// Clean up the default schema under the catalog if creation failed

I believe this scenario is only used for kafka catalog currently, so can you add more information about it?

Reply:
This has nothing to do with the Kafka catalog, the comment has been updated